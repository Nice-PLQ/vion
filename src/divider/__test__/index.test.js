import { shallowMount } from '@vue/test-utils';
import Divider from '../divider.vue';

describe('Divider', () => {
  const wrapper = shallowMount(Divider);

  test('类型', async () => {
    await wrapper.setProps({ type: 'bold' });
    expect(wrapper.classes()).toContain('v-divider_bold');
    expect(wrapper.element).toMatchSnapshot();
  });

  test('高度', async () => {
    await wrapper.setProps({ height: '48px' });
    expect(wrapper.element.style.height).toBe('48px');
  });

  test('颜色', async () => {
    await wrapper.setProps({ color: '#B0B3BF' });
    expect(wrapper.element.style.backgroundColor).toBe('rgb(176, 179, 191)');
  });
});
