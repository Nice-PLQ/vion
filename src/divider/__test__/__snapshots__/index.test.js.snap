// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider 类型 1`] = `
<div
  class="v-divider v-divider_bold"
/>
`;
